charts/ingress-nginx-controller/values.yaml

```diff
@@ -0,0 +1,44 @@
+# the following defaults apply to an on-prem bare-metal setup in the same spirit as the
+# older similarly named wrapper chart 'nginx-ingress-controller' (note the swapped words
+# 'nginx' and 'ingress') We assume no load balancer support and instead expose NodePorts
+# on ports 31773 and 31772, assuming traffic gets to these ports in another way.
+#
+# See https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/values.yaml for all possible values to override.
+ingress-nginx:
+  controller:
+    enableTopologyAwareRouting: true
+    # -- Use a `DaemonSet` or `Deployment`
+    kind: DaemonSet
+    service:
+      type: NodePort  # or LoadBalancer
+      externalTrafficPolicy: Local
+      nodePorts:
+        # The nginx instance is exposed on ports 31773 (https) and 31772 (http)
+        # on the node on which it runs. You should add a port-forwarding rule
+        # on the node or on the loadbalancer that forwards ports 443 and 80 to
+        # these respective ports.
+        https: 31773
+        http: 31772
```

Comment:
A more opportunistic default here would be to use `type: Deployment` and `type: LoadBalancer`, and document how to configure this if you have to resort to NodePort.

The NodePort approach always requires manual configuration of some external load balancer/firewall to round-robin between node IPs and is error-prone. It's also a bit annoying to have to decide on some global ports that may not be used otherwise.

Most managed K8s clusters have support for LoadBalancers, you can also get this for your own clusters in hcloud etc. It's even possible to do it for pure bare metal, without any "load balancer hardware", by using BGP or some leadership election over who's announcing the "load balancer ip" via ARP (https://metallb.universe.tf/configuration/_advanced_l2_configuration/).

Reply:
I swapped the defaults as you suggested and documented how to get the previous behaviour back.